microprofile/tests/junit5/src/main/java/io/helidon/microprofile/tests/junit5/HelidonJunitExtension.java

```diff
@@ -458,6 +460,43 @@ private AddBeansExtension(Class<?> testClass, List<AddBean> addBeans) {
 
         @SuppressWarnings("unchecked")
         void registerOtherBeans(@Observes AfterBeanDiscovery event) {
+
+            Class<?> aClass = testClass;
+            HashMap<Annotation, String> socketAnnotations = new HashMap<>();
+            while (aClass != null) {
+                for (Field field : aClass.getDeclaredFields()) {
+                    if (field.isAnnotationPresent(Inject.class) && field.isAnnotationPresent(Socket.class)) {
+                        field.setAccessible(true);
+                        socketAnnotations.put(field.getAnnotation(Socket.class), field.getAnnotation(Socket.class).value());
+                    }
+                }
+                aClass = aClass.getSuperclass();
+            }
+
+            //register for all named Ports
+            socketAnnotations.forEach((socket, namedPort) -> {
+
+                Client namedClient = ClientBuilder.newClient();
```

Comment:
Also you do not want to create a new client for each socket - client does not care about endpoints, so you can reuse the same client as is used for default socket.

Reply:
Fixed